File: src/1191-K-Concatenation-Maximum-Sum/1191.go
func kConcatenationMaxSum(arr []int, k int) int {
    mod, cur, res, total := 1000000007, 0, 0, 0
    for _, v := range arr {
        total += v
    }
    
    for i := 0; i < len(arr)*min(k, 2); i++ {
        cur = max(cur, 0) + arr[i%len(arr)]
        res = max(res, cur)
    }
    return max(max(res, 0), ((k - 2)*total + res)%mod)
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: src/1191-K-Concatenation-Maximum-Sum/1191.js
var kConcatenationMaxSum = function(arr, k) {
    var mod = 1e9 + 7, cur = 0, res = 0, total = 0;
    for (let num of arr) total += num;
    
    for (var i = 0; i < arr.length*Math.min(k, 2); ++i) {
        cur = Math.max(cur, 0) + arr[i%arr.length];
        res = Math.max(res, cur);
    }
    return Math.max(0, res, ((k-2)*total + res)%mod);
};